code/modules/mob/mob_movespeed.dm

```diff
@@ -0,0 +1,96 @@
+
+/*Current movespeed modification list format: list(id = list(
+	priority,
+	legacy slowdown/speedup amount,
+	))
+*/
+
+//ANY ADD/REMOVE DONE IN UPDATE_MOVESPEED MUST HAVE THE UPDATE ARGUMENT SET AS FALSE!
+/mob/proc/add_movespeed_modifier(id, update = TRUE, priority = 0, flags = NONE, override = FALSE, legacy_slowdown = 0)
+	var/list/temp = list(priority, flags, legacy_slowdown)			//build the modification list
+	if(LAZYACCESS(movespeed_modification, id))
+		if(movespeed_modifier_identical_check(movespeed_modification[id], temp))
+			return FALSE
+		if(!override)
+			return FALSE
+		else
+			remove_movespeed_modifier(id, update)
+	LAZYSET(movespeed_modification, id, list(priority, flags, legacy_slowdown))
+	if(update)
+		update_movespeed(TRUE)
+	return TRUE
+
+/mob/proc/remove_movespeed_modifier(id, update = TRUE)
+	if(!LAZYACCESS(movespeed_modification, id))
+		return FALSE
+	LAZYREMOVE(movespeed_modification, id)
+	UNSETEMPTY(movespeed_modification)
+	if(update)
+		update_movespeed(FALSE)
+	return TRUE
+
+/mob/proc/has_movespeed_modifier(id)
+	return LAZYACCESS(movespeed_modification, id)
+
+/mob/proc/update_config_movespeed()
+	add_movespeed_modifier(MOVESPEED_ID_CONFIG_SPEEDMOD, FALSE, 100, override = TRUE, legacy_slowdown = get_config_multiplicative_speed())
+
+/mob/proc/get_config_multiplicative_speed()
+	if(!islist(GLOB.mob_config_movespeed_type_lookup) || !GLOB.mob_config_movespeed_type_lookup[type])
+		return 0
+	else
+		return GLOB.mob_config_movespeed_type_lookup[type]
+
+/mob/proc/update_movespeed(resort = TRUE)
+	if(resort)
+		sort_movespeed_modlist()
+	. = 0
+	for(var/id in get_movespeed_modifiers())
+		var/list/data = movespeed_modification[id]
+		. += data[MOVESPEED_DATA_INDEX_LEGACY_SLOWDOWN]
+	cached_legacy_slowdown = .

```

Comment:
why are you calling it legacy_slowdown?

Reply:
what's legacy about it?

Reply:
also if it's legacy why aren't you removing it?

Reply:
this is the current version of slowdown, which will be the "old" slowdown when pixel movement is out, but in reality I'm going to rename it to multiplicative slowdown